bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java

```diff
@@ -317,4 +321,31 @@ public int getRollerFav(int id) {
         }
         return -1;
     }
+
+    public static String extractFwVersion(@Nullable String version) {
+        if (version != null) {
+            if (version.chars().filter(ch -> ch == '-').count() >= 3) {
+                // new format starting with version 1.0
+                // e.g. 20210226-091047/v1.10.0-rc2-89-g623b41ec0-master
+            }
+            if (version.contains("@")) { // standard format
+                return substringBetween(getString(version), "/", "@");
+            }
+
+            if (version.indexOf('/') != version.lastIndexOf('/')) {
+                // includes 2 /, e.g. beta
+                return substringAfterLast(version, "/");
+            }
+        }
+        return "";
+    }
```

Comment:
It seems more reasonable to extract the version by using regular expressions. That would increase code maintainability.

Reply:
I not familiar with regular expressions, would need help

Reply:
It's not that difficult. You could take a look at other bindings: https://github.com/openhab/openhab-addons/search?q=pattern.compile

Reply:
yup, I know how this works in general, but finding the correct pattern...
I found this page: https://regex101.com, realy nice, allows you to test online, but also decodes a regex and shows the logic

Implemented regex search